### PR TITLE
Add scale_voltage_current_power to ModelChain.pvwatts_dc

### LINKED ISSUE #476: document or support modules_per_string strings_per_inverter with pvwatts in modelchain
Hi, 

I am trying to run Modelchain with pvwatt model but it seems that the `modules_per_string` and `strings_per inverter ` doesn't have any affect on the total output. 

I am not sure why is it so. 
May be ModelChain isn't supporting so. If that's the case how can I achieve the desired result?

Here is my code: 

Thanks in advance
```
# built-in python modules
import os
import inspect

# scientific python add-ons
import numpy as np
import pandas as pd

# plotting stuff
# first line makes the plots appear in the notebook
%matplotlib inline 
import matplotlib.pyplot as plt
import matplotlib as mpl
# seaborn makes your plots look better
try:
    import seaborn as sns
    sns.set(rc={"figure.figsize": (12, 6)})
    sns.set_color_codes()
except ImportError:
    print('We suggest you install seaborn using conda or pip and rerun this cell')

# finally, we import the pvlib library
import pvlib

tmy = pd.read_csv("http://re.jrc.ec.europa.eu/pvgis5/tmy.php?lat=29.74&lon=40.10")

tmy.Date = pd.to_datetime(tmy.Date, format='%Y-%d-%m %H:%M:%S')

tmy.rename(columns={' Ghor':'ghi','Dhor':'dhi','DNI':'dni','Tair':'temp_air',
                        'Ws':'wind_speed'},inplace=True)					
						
tmy.set_index(tmy['Date'],inplace=True)
    #Drop unnecessary column

tmy = tmy.drop('Date', 1)
tmy = tmy.drop('RH', 1)
tmy = tmy.drop('IR', 1)
tmy = tmy.drop(' Wd', 1)
tmy = tmy.drop('Pres', 1)

#module =Jinko_Solar_JKM320P_72_V
#inverter = ABB__PVS980_58_2000kVA_K__N_A_V__CEC_2018_

lat = 29.74
lon = 40.10
altitude = 676
tz = 'Etc/GMT+3'  

loc = pvlib.location.Location(latitude=lat,longitude= lon,tz=tz)

#model = pvwatts 
pvwatts_system = pvlib.pvsystem.PVSystem(module_parameters={'pdc0': 320, 'gamma_pdc': -0.0041},inverter_parameters={'pdc' : 3200000, 'pdc0' : 2024292, 'eta_inv_nom':0.988, 'eta_inv_ref':0.986},surface_tilt = 20, surface_azimuth=0,
                    modules_per_string=30,strings_per_inverter=267, albedo = 0.2)
					
mc = pvlib.modelchain.ModelChain(pvwatts_system, loc, transposition_model ="perez",aoi_model = 'ashrae',spectral_model='no_loss')
print(mc)
mc.run_model(times=tmy.index,weather=tmy)
a = mc.ac
a = pd.Series.to_frame(a)
a = a * 530  # 530 = number of inverters in the system 

a['month'] = a.index
a.month = a.month.dt.month
monthly = a.groupby('month').sum()
```

### PR DESCRIPTION
**Pull request by cwhanse**
- [x] Closes #476 
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - [x] Updates entries to [`docs/sphinx/source/api.rst`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst) for API changes.
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

Adds scaling by modules per string and strings per inverter to the ModelChain.pvwatts_dc method. This behavior parallels ModelChain.sapm and ModelChain.singlediode. User code is not likely to be affected since PVSystem.modules_per_strings and PVSystem.strings_per_inverter both default to 1.

### CODE REVIEW

**Comment by cwhanse** [comment]
@wholmgren I should have looked at the base level function. `pvlib.pvsystem.scale_voltage_current_power` expects the input DataFrame to contain voltage, current and power. We could make the existing function more flexible, or add a new function that only scales a Series input of power. Preference?

**Comment by wholmgren** [comment]
I had the same oversight despite the fact that "voltage" and "current" are in the name. I'd prefer a new function over changing the existing function. Maybe this is a little lazy and inconsistent, but I'd also be ok with skipping the function layer and only implementing the `PVSystem` method or just putting it in the `ModelChain.pvwatts_dc` method.

**Comment by cwhanse** [comment]
I got impatient and tried the first option (make existing function more general).  One drawback of the existing function is that it requires all keys including 'i_x' and 'i_xx' which are produced by the singlediode function probably to be consistent with the output of the sapm.

**Comment by cwhanse** [comment]
@wholmgren see what you think before I undo or fix what is broken. I think we can keep the same signature of `scale_voltage_current_power`.

**Comment by wholmgren** [comment]
You've managed to make the name and signature make sense, but using the seldom-used `name` attribute as a switch feels wrong. 

Another option to consider:
* Within `ModelChain.pvwatts_dc` we create a temporary `DataFrame` with `p_mp=pdc`, `v_mp=pdc`, and `i_mp=1`.
* Within `scale_voltage_current_power`, use one of the more flexible pandas column selection methods
* Within `ModelChain.pvwatts_dc` set the result equal to the appropriate column so just the scaled pdc Series is retained.

**Comment by cwhanse** [comment]
If `name` is to be avoided for Series I don't see how to use this function to e.g. scale a voltage Series. In that case I think the temporary DataFrame is the way to go.

File: pvlib/pvsystem.py [hunk 1]
```diff
@@ -2648,13 +2648,16 @@ def scale_voltage_current_power(data, voltage=1, current=1):
...
     # could make it work with a dict, but it would be more verbose
+    voltage_keys = ['v_mp', 'v_oc']
+    current_keys = ['i_mp', 'i_x', 'i_xx', 'i_sc']
+    power_keys = ['p_mp']
     data = data.copy()
-    voltages = ['v_mp', 'v_oc']
-    currents = ['i_mp', 'i_x', 'i_xx', 'i_sc']
+    voltages = voltage_keys and data.columns
+    currents = current_keys and data.columns
+    powers = power_keys and data.columns
```
Comment (wholmgren) [inline pvlib/pvsystem.py:2657]: I was thinking of something like this:

```python
In [28]: voltage = 5

In [29]: current = 10

In [30]: df = pd.DataFrame([[1, 1, 1]], columns=['p_mp', 'v_mp', 'i_mp'])

In [31]: df
Out[31]:
   p_mp  v_mp  i_mp
0     1     1     1

In [32]:     voltage_keys = ['v_mp', 'v_oc']
    ...:     current_keys = ['i_mp', 'i_x', 'i_xx', 'i_sc']
    ...:     power_keys = ['p_mp']

In [33]: df.filter(voltage_keys, axis=1) * voltage
Out[33]:
   v_mp
0     5

In [34]: df.filter(current_keys, axis=1) * current
Out[34]:
   i_mp
0    10
```

Comment (wholmgren) [reply pvlib/pvsystem.py]:
```
In [35]: voltage_df = df.filter(voltage_keys, axis=1) * voltage

In [36]: current_df = df.filter(current_keys, axis=1) * current

In [37]: df = pd.concat([voltage_df, current_df], axis=1)

In [38]: df['p_mp'] = df['v_mp'] * df['i_mp']

In [39]: df
Out[39]:
   v_mp  i_mp  p_mp
0     5    10    50
```

Comment (wholmgren) [reply pvlib/pvsystem.py]: I don't know how to scale anything other than `p_mp` without hugely complicating the code. I also don't know that there's a need for it at this time.

Comment (cwhanse) [reply pvlib/pvsystem.py]: The `pvwatts` methods don't produce voltage or current.

Comment (wholmgren) [reply pvlib/pvsystem.py]: I was still in the mindset that we could provide a DataFrame with `p_mp=pdc`, `v_mp=pdc`, and `i_mp=1` as I suggested above. But yes it is better to just multiply the power keys by the voltage and current scale factors.

Comment (cwhanse) [reply pvlib/pvsystem.py]: One issue with the temporary DataFrame: when converting back to Series, the column label `'p_mp'` is assigned to the Series name, which affects testing but likely no user code. Any concern here?

Comment (wholmgren) [reply pvlib/pvsystem.py]: no concern. pandas `assert_series_equal` only recently started checking that attribute.

**Comment by cwhanse** [comment]
Test failure is unrelated, timeout when fetching SURFRAD data.

File: pvlib/pvsystem.py [hunk 2]
```diff
@@ -2648,14 +2648,13 @@ def scale_voltage_current_power(data, voltage=1, current=1):
...
-    data['p_mp'] *= voltage * current
-
-    return data
+    voltage_keys = ['v_mp', 'v_oc']
+    current_keys = ['i_mp', 'i_x', 'i_xx', 'i_sc']
+    power_keys = ['p_mp']
+    voltage_df = data.filter(voltage_keys, axis=1) * voltage
+    current_df = data.filter(current_keys, axis=1) * current
+    power_df = data.filter(power_keys, axis=1) * voltage * current
+    return pd.concat([voltage_df, current_df, power_df], axis=1)
```
Comment (wholmgren) [inline pvlib/pvsystem.py:2657]:
```suggestion
    df = pd.concat([voltage_df, current_df, power_df], axis=1)
    df_sorted = df[data.columns]  # retain original column order
    return df_sorted
```

File: docs/sphinx/source/whatsnew/v0.9.0.rst [hunk 3]
```diff
@@ -83,6 +83,10 @@ Enhancements
   automatically switch to using ``'effective_irradiance'`` (if available) for
   cell temperature models, when ``'poa_global'`` is not provided in input
   weather or calculated from input weather data.
+* :py:meth:`~pvlib.modelchain.ModelChain.pvwatts_dc` now scales the DC power
+  by `pvsystem.PVSystem.modules_per_strings` and
+  `pvsystem.PVSystem.strings_per_inverter`. Note that both attributes still
```
Comment (wholmgren) [inline docs/sphinx/source/whatsnew/v0.9.0.rst:88]:
```suggestion
  by ``pvsystem.PVSystem.modules_per_strings`` and
  ``pvsystem.PVSystem.strings_per_inverter``. Note that both attributes still
```

**Review by wholmgren — APPROVED** [review]

**Comment by wholmgren** [comment]
failures are still surfrad io. 

Thanks @cwhanse!

**Pull request merged**